### ISSUE TITLE
Predict noise labels

### ISSUE THREAD

**Issue by carlogarro** [PaulAlbert31/LabelNoiseCorrection#5]
Hello, I would like to know if using the beta model at the end of the training could be used to detect all the noise labels or those are separated along with the training, and the beta model, in the end, is only useful to separate the hardest ones?

**Comment by EricArazo** [comment]
Hello Carlogarro, the idea of the beta model is to associate a probability to each sample. That probability depends on the loss value of each sample and during training is used to guide the training loss; we do not discard any sample during training. You could use the beta model at the end of the training to identify the noisy samples, but this is not its purpose in this paper.

Please let me know if you have any other questions.

**Comment by carlogarro** [comment]
Thanks, Eric,
One more question, could you explain to me both methods in BetaMixture1D class?
create_lookup(y) and predict(x) and why when doing create_lookup you pass 1 as an argument.

Thank you again!!

**Comment by EricArazo** [comment]
Hello Carlogarro,

The create_lookup(y) method creates a lookup table that is a list that contains the probability values given by the BMM for each loss value (1 corresponding to 100% prob of being noisy). Then, when the table is created, to compute the probabilities corresponding to the component with higher value, you have to pass one as an argument (the component 0 corresponds to the clean samples and the component 1 to the noisy).

The method predict(), given a loss value, returns True or False if the sample is noisy or clean. 

Best,
Eric
